auth.py

```diff
@@ -0,0 +1,154 @@
+from os.path import expanduser
+import configparser
+import boto.sts
+import boto.s3
+import base64
+import xml.etree.ElementTree as ET
+import getpass
+import time
+import os
+import asyncio
+from pyppeteer import launch
+
+async def main():
+    # region: The default AWS region that this script will connect
+    # to for all API calls
+    region = os.environ.get('AWS_REGION')
+
+    # output format: The AWS CLI output format that will be configured in the
+    # saml profile (affects subsequent CLI calls)
+    outputformat = os.environ.get('AWS_OUTPUT_FORMAT')
+
+    # awsconfigfile: The file where this script will store the temp
+    # credentials under the saml profile
+    awsconfigfile = '/.aws/credentials'
+
+    # Load or create AWS config file
+    home = expanduser("~")
+    filename = home + awsconfigfile
+
+    # Read in the existing config file
+    config = configparser.ConfigParser()
+    config.read(filename)
+
+    # If aws config doesn't exist, create one because boto requires it
+    if not config.has_section('default'):
+        config.add_section('default')
+        config.set('default', 'output', outputformat)
+        config.set('default', 'region', region)
+        config.set('default', 'aws_access_key_id', '')
+        config.set('default', 'aws_secret_access_key', '')
+        # Write the updated config file
+        with open(filename, 'w+') as configfile:
+            config.write(configfile)
+
+    browser = await launch(
+        headless=True,
+        executablePath="/usr/bin/chromium-browser",
+        args=['--no-sandbox', '--disable-gpu']
+    )
+    page = await browser.newPage()
+    await page.goto('http://bu.edu/awslogin')
```

Comment:
`https://www.bu.edu/awslogin` ?

Reply:
Actually, I want to make it a standalone parameter. And yes, I'll make it the one you suggest. Thanks!